### PR TITLE
Adam with weight decay per parameter + some activation functions

### PR DESCRIPTION
**Pull request by efylmzr**
Some activation functions which are helpful for neural network training such as Sigmoid, Softplus and Swish. The AdamW is extended to handle per parameter weight decays.

### CODE REVIEW

**Review by wjakob — CHANGES_REQUESTED** [review]
Hi Ekrem -- thank you for the PR. Here are some comments from me.

File: drjit/opt.py [hunk 1]
```diff
@@ -1146,7 +1146,7 @@ def __init__(
         beta_1: float = 0.9,
         beta_2: float = 0.999,
         epsilon: float = 1e-8,
-        weight_decay: float = 0.01,
+        weight_decay: Optional[float | Mapping[str, float]] = None,
```
[the remark this answers is not in the document]
Comment (efylmzr) [reply drjit/opt.py:1149]: I did not get your comment here. The other optimizers do not have any access to the `weight_decay` parameter. I just have an additional dictionary in `AdamW` and increase the size of the `tuple` (`extra`) in`set_weight_decay` function. Can you elaborate your comment about keeping everything in `AdamW`?

[the remark this answers is not in the document]
Comment (wjakob) [reply drjit/opt.py]: Sorry, I misread the diff and had thought that you were modifying the base optimizer. I think it would be better if the `set_weight_decay` function works like the `set_learning_rate` function, please check out the docstring of this function. Right now, all of the optimizers are written so that they don't use key-based dictionary lookups in `opt.step()`. They iterate over dictionaries but never explicitly search a dictionary for a string-based key (which involves string hashing etc.)
